example/mongo-driver-v2/go.mod

```
@@ -0,0 +1,29 @@
+module github.com/instana/go-sensor/example/mongo-driver-v2
+
+go 1.23.6
```

```suggestion
go 1.23
```

Aligning with our minimum supported version.